Automattic/wp-calypso
# Jetpack Credentials: Refactor credentials test logic to make it reusable

We require to prevent site restores if Jetpack credentials are not valid. For this purpose, @villanovachile made an important change in #66468, implementing a new API to test credentials on Jetpack Credentials Setting page.

The idea of this PR is to move the credentials test logic to a reusable approach across components, so it can be used in Jetpack Settings and also on Backup/Restore Flow. The implementation for Backup/Restore Flow will be handled in a separate PR.

#### Proposed Changes
* Move the API call logic to a data-layer and create the necessary actions, reducers and selectors to handle the new state.
* Refactor `AdvancedCredentials` component to use this new logic.
  * If a site has no credentials, mark them as invalid, so we don't need to call the API in those cases.
  * If we update site credentials, mark them as valid, as the update process test the credentials as part of its procedure.

#### Testing Instructions


_(reusing #66468 testing instructions as this refactor impacts the same functionalities)_
* Spin up a new Jurassic Ninja site (or use an existing self hosted site), connect Jetpack, and add a Jetpack Backup plan
* Add server credentials at cloud.jetpack.com > Settings, or follow the steps here: https://jetpack.com/support/backup/backups-via-the-jetpack-plugin/adding-credentials-to-jetpack/
* Navigate to cloud.jetpack.com > Settings, or to WPCOM > Settings > Jetpack, and confirm that it shows connected (results should match test from RWDB)
* SSH into your site and change password, then navigate back to cloud.jetpack.com > Settings or to WPCOM > Settings > Jetpack, and Calypso should show Disconnected prompting user to re-enter credentials

#### Pre-merge Checklist






Related to #66468 & #67561
